kratos/utilities/derivatives_extension.h

```diff
@@ -0,0 +1,166 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:   Suneth Warnakulasuriya
+//
+//
+
+#if !defined(KRATOS_DERIVATIVES_EXTENSION_INCLUDED)
+#define KRATOS_DERIVATIVES_EXTENSION_INCLUDED
+
+// System includes
+#include <iosfwd>
+#include <vector>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/dof.h"
+#include "includes/process_info.h"
+#include "utilities/indirect_scalar_fwd.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/**
+ * @class DerivativesExtension
+ * @ingroup KratosCore
+ * @brief Interface extensions for elements and conditions.
+ */
+class DerivativesExtension
```

Comment:
this will be really useful from how I understand, thx!

Reply:
:+1:

Reply:
Can you add Doxygen documentation?